napari_plugin_manager/qt_plugin_dialog.py

```diff
@@ -1027,26 +1073,33 @@ def _add_items(self):
                 self._add_items_timer.stop()
             return
 
-        data = self._plugin_data.pop(0)
-        metadata, is_available_in_conda, extra_info = data
-        display_name = extra_info.get('display_name', metadata.name)
-        if metadata.name in self.already_installed:
-            self.installed_list.tag_outdated(metadata, is_available_in_conda)
-        else:
-            if metadata.name not in self.available_set:
-                self.available_set.add(metadata.name)
-                self.available_list.addItem(
-                    ProjectInfoVersions(
-                        metadata,
-                        display_name,
-                        extra_info['pypi_versions'],
-                        extra_info['conda_versions'],
-                    )
+        for _ in range(2):
```

Comment:
Could we make the `2` a variable, like `batch_size` so it's better documented? and update the docstring (it's not one by one, but in batches)
Not sure it makes sense to make it a kwarg, but just so it's not a random integer.

For the record I tried 10 on my machine it made scrolling laggy--not good.
`3` felt pretty good though (2020 M1), but we should target the lower common denominator.

Reply:
Indeed, what you describe was exactly what I found, 2 is a sweet spot in this case.